pkg/lib/cockpit.js

```diff
@@ -4263,16 +4263,16 @@ function factory() {
         }
 
         self.allowed = null;
-        self.user = options ? options.user : null; // pre-fill for unit tests
-        self.is_superuser = options ? options._is_superuser : null; // pre-fill for unit tests
+        self.user = options?.user ?? null; // pre-fill for unit tests
+        self.is_superuser = options?._is_superuser ?? null; // pre-fill for unit tests
```

Comment:
> unit test failures/hangs, this broke something. Are you set up for running `make check` with the browser tests? Do you want to figure this our yourself, or want some help?

Honestly, I think I need some guidance on this matter. Are there any resources on cockpit testing other than the `HACKING.md` page. I've tried `make check` and `./test-server` but they only show that the tests fail, without any pointers to what could be the cause. Is there a way to do that?

Reply:
And how exactly did you figure out that this is what breaks `test-permissions.html`?

Reply:
I reverted parts of your patch until I found the right one. I knew it was related to permissions, so I started with the related ones (but without that knowledge, bisecting is not much slower).

Reply:
"Normal" test failures have some verbose output. It could be that the test hangs, then it's more difficult. ./test-server works better then, as it gives you live output, and you can also see the browser console.

A good approach in this case is to just back out problematic changes, and land the 95% good ones. The the remainder is either much easier to investigate, or not important enough to care.